hplsql/src/main/java/org/apache/hive/hplsql/Cmp.java

```diff
@@ -138,28 +147,26 @@ else if (query2.error()) {
       exec.signal(query2);
       return null;
     }
-    ResultSet rs1 = query1.getResultSet();
-    ResultSet rs2 = query2.getResultSet();
-    if (rs1 == null || rs2 == null) {
+    if (query1 == null || query2 == null) {
       exec.setSqlCode(-1);
       return null;
     }
     boolean equal = true;
     tests = 0;
     failedTests = 0;
     try {
-      ResultSetMetaData rm1 = rs1.getMetaData();
-      ResultSetMetaData rm2 = rs2.getMetaData();
-      int cnt1 = rm1.getColumnCount();
-      int cnt2 = rm2.getColumnCount();
+      Metadata rm1 = query1.metadata();
+      Metadata rm2 = query2.metadata();
+      int cnt1 = rm1.columnCount();
+      int cnt2 = rm2.columnCount();
       tests = cnt1;
-      while (rs1.next() && rs2.next()) {
+      while (query1.next() && query2.next()) {
         for (int i = 1; i <= tests; i++) {
```

Comment:
Yes, I missed this. It should have been 0 based. Fixed it. I also removed the 2 threads since Tez doesn't support concurrent queries within the same session.